x-pack/plugins/rule_registry/server/alert_data_client/tests/bulk_update.test.ts

```diff
@@ -0,0 +1,458 @@
+          })
+        ).rejects.toThrowErrorMatchingInlineSnapshot(`"Unauthorized for fake.rule and apm"`);
+
+        expect(auditLogger.log).toHaveBeenLastCalledWith({
+          message: `Failed attempt to update alert [id=${fakeAlertId}]`,
+          event: {
+            action: 'alert_update',
+            category: ['database'],
+            outcome: 'failure',
+            type: ['change'],
+          },
+          error: {
+            code: 'Error',
+            message: 'Unauthorized for fake.rule and apm',
+          },
+        });
+      });
+
+      test('logs multiple error events in audit logger', async () => {
+        const indexName = '.alerts-observability-apm.alerts';
+        const alertsClient = new AlertsClient(alertsClientParams);
+        esClientMock.mget.mockResolvedValueOnce(
+          elasticsearchClientMock.createApiResponse({
+            body: {
+              docs: [
+                {
+                  _id: successfulAuthzHit,
+                  _index: indexName,
+                  _source: {
+                    [RULE_ID]: 'apm.error_rate',
+                    [ALERT_OWNER]: 'apm',
+                    [ALERT_STATUS]: 'open',
+                    [SPACE_IDS]: [DEFAULT_SPACE],
+                  },
+                },
+                {
+                  _id: unsuccessfulAuthzHit,
+                  _index: indexName,
+                  _source: {
+                    [RULE_ID]: fakeRuleTypeId,
+                    [ALERT_OWNER]: 'apm',
+                    [ALERT_STATUS]: 'open',
+                    [SPACE_IDS]: [DEFAULT_SPACE],
+                  },
+                },
+              ],
+            },
+          })
+        );
+
+        await expect(
+          alertsClient.bulkUpdate({
+            ids: [successfulAuthzHit, unsuccessfulAuthzHit],
+            query: undefined,
+            index: indexName,
+            status: 'closed',
+          })
+        ).rejects.toThrowErrorMatchingInlineSnapshot(`"Unauthorized for fake.rule and apm"`);
+        expect(auditLogger.log.mock.calls).toHaveLength(2);
+        expect(auditLogger.log.mock.calls[0][0]).toEqual({
+          message: `Failed attempt to update alert [id=${unsuccessfulAuthzHit}]`,
+          event: {
+            action: 'alert_update',
+            category: ['database'],
+            outcome: 'failure',
+            type: ['change'],
+          },
+          error: {
+            code: 'Error',
+            message: 'Unauthorized for fake.rule and apm',
+          },
+        });
+        expect(auditLogger.log.mock.calls[1][0]).toEqual({
+          message: `Failed attempt to update alert [id=${successfulAuthzHit}]`,
+          event: {
+            action: 'alert_update',
+            category: ['database'],
+            outcome: 'failure',
+            type: ['change'],
+          },
+          error: {
+            code: 'Error',
+            message: 'Unauthorized for fake.rule and apm',
+          },
+        });
```

Comment:
Optional nit: Jest provides specific functions to deal with these situations, e.g.,

```ts
expect(auditLogger.log).toHaveBeenCalledTimes(2);
expect(auditLogger.log).toHaveBeenNthCalledWith(
  1,
  {...}
);
expect(auditLogger.log).toHaveBeenNthCalledWith(
  2,
  {...}
);
```

Reply:
Thanks! Fixed here: https://github.com/elastic/kibana/pull/106297/commits/ba1857efe6747f0eae50853ac4869a762836e253